File: python/Prog_붕대감기.py
def solution(bandage, health, attacks):
    need_time, plus_per_seconds, plus_success = bandage
    hp = health
    max_hp = health

    last_attack_time = attacks[-1][0]
    attack = [0 for _ in range(last_attack_time+1)]
    
    for t,d in attacks :
        attack[t] = d
    
    success_cnt = 0
    for t in range(last_attack_time+1) :
        if attack[t] :
            hp -= attack[t]
            if hp <= 0 : return -1
            success_cnt = 0
        else :
            success_cnt += 1
            hp = min(max_hp, hp + plus_per_seconds)
            if success_cnt == need_time :
                success_cnt = 0
                hp = min(max_hp, hp + plus_success)
    
    return hp

File: java/src/Prog_붕대감기.java
class Solution {
    public int solution(int[] bandage, int health, int[][] attacks) {
        int cur_attack_idx = 0;
        int max_health = health;
        
        int tmp = 0;
        for (int time=0;time<=attacks[attacks.length-1][0];time++) {
            if (time == attacks[cur_attack_idx][0]) {
                health -= attacks[cur_attack_idx][1];
                if (health <= 0) return -1;
                cur_attack_idx++;
                tmp = 0;
            } else {
                health = Math.min(max_health, bandage[1] + health);
                tmp ++;
                
                if (tmp == bandage[0]) {
                    health = Math.min(max_health, bandage[2] + health);
                    tmp = 0;
                }
            }
        }
        
        return health;
    }
}